src/analyze/analyze-chid.c

```diff
@@ -223,6 +247,47 @@ static int smbios_fields_acquire(char16_t *fields[static _CHID_SMBIOS_FIELDS_MAX
         return 0;
 }
 
+static int edid_parse(const char *edid_path, char16_t **ret_panel) {
+        int r;
+        _cleanup_free_ char *edid_content = NULL;
+        size_t edid_size = 0;
+
+        r = read_full_file(edid_path, &edid_content, &edid_size);
+        if (r < 0)
+                return r;
+
+        EdidHeader header;
+        if (!edid_parse_blob(edid_content, edid_size, &header))
+                return -EINVAL;
+
+        EdidPanelId panel_id;
+        if (!edid_get_panel_id(&header, &panel_id))
+                return -EINVAL;
+
+        *ret_panel = new0(char16_t, 5);
+        edid_panel_id(&panel_id, *ret_panel);
+        return 0;
+}
+
+static int edid_acquire(char16_t **ret_panel) {
+        _cleanup_closedir_ DIR *dir = opendir("/sys/class/drm");
```

Comment:
would be nicer to go via sd-device api. see manager_count_external_displays() in src/login/logind-core.c for an example.

then use sd_device_get_sysattr_value() to read the edid data